e2e/machineset_migration_test.go

```diff
@@ -213,13 +214,676 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
+						HaveField("Status", Equal(corev1.ConditionTrue)),
+						HaveField("Reason", Equal("AuthoritativeAPINotMachineAPI")),
+					))),
+				))
+			})
+
+			It("should switch AuthoritativeAPI to MachineAPI succeed", func() {
+				updateMachineSetAuthoritativeAPI(mapiMachineSet, mapiv1beta1.MachineAuthorityMachineAPI, mapiv1beta1.MachineAuthorityMachineAPI)
+				verifySynchronizedCondition(mapiMachineSet, mapiv1beta1.MachineAuthorityMachineAPI)
+				verifyMAPIPausedCondition(mapiMachineSet, mapiv1beta1.MachineAuthorityMachineAPI)
+				verifyCAPIPausedCondition(capiMachineSet, mapiv1beta1.MachineAuthorityMachineAPI)
+			})
+
+			It("should scale up MAPI MachineSet to 3 succeed after switching AuthoritativeAPI to MachineAPI", func() {
+				By("Scaling up MAPI MachineSet to 3 replicas")
+				Expect(mapiframework.ScaleMachineSet(mapiMSAuthCAPIName, 3)).To(Succeed(), "should be able to scale up MAPI MachineSet")
+				Eventually(komega.Object(mapiMachineSet)).Should(HaveField("Spec.Replicas", HaveValue(Equal(int32(3)))))
+
+				By("Verifying a new MAPI Machine is created and Paused condition is False")
+				Eventually(func() (*mapiv1beta1.Machine, error) {
+					mapiMachine, err := mapiframework.GetLatestMachineFromMachineSet(ctx, cl, mapiMachineSet)
+					if err != nil {
+						return nil, err
+					}
+					return mapiMachine, nil
+				}, capiframework.WaitLong, capiframework.RetryLong).Should(SatisfyAll(
+					HaveField("Status.Conditions", ContainElement(SatisfyAll(
+						HaveField("Type", Equal(MAPIPausedCondition)),
+						HaveField("Status", Equal(corev1.ConditionFalse)),
+					))),
+					HaveField("Status.AuthoritativeAPI", Equal(mapiv1beta1.MachineAuthorityMachineAPI)),
+					HaveField("Status.Phase", HaveValue(Equal(string(mapiv1beta1.PhaseRunning)))),
+				))
+
+				By("Verifying there is a mirrored paused CAPI Machine")
+				Eventually(func(g Gomega) {
+					capiMachine, err := capiframework.GetNewestMachineFromMachineSet(cl, capiMachineSet)
+					g.Expect(err).ToNot(HaveOccurred(), "error getting CAPI machine")
+					g.Expect(capiMachine.Status.Conditions).ToNot(BeEmpty(), "CAPI Machine should have conditions")
+
+					var pausedConditionFound bool
+					for _, cond := range capiMachine.Status.Conditions {
+						if string(cond.Type) == string(MAPIPausedCondition) && cond.Status == corev1.ConditionTrue {
+							pausedConditionFound = true
+							break
+						}
+					}
+					g.Expect(pausedConditionFound).To(BeTrue(), "Expected Paused=True condition in CAPI Machine")
+				}, capiframework.WaitShort, capiframework.RetryShort)
+
+				By("Verifying old Machines still exist and authority on them is still ClusterAPI")
+				Eventually(komega.Object(firstMAPIMachine)).Should(HaveField("Status.AuthoritativeAPI", Equal(mapiv1beta1.MachineAuthorityClusterAPI)))
+				Eventually(komega.Object(secondMAPIMachine)).Should(HaveField("Status.AuthoritativeAPI", Equal(mapiv1beta1.MachineAuthorityClusterAPI)))
```

Comment:
Should we put timeouts on these eventually calls? So we stay consistent thoughout the file?